### PR TITLE
Added fuzzytester

### PR DESCRIPTION
**Pull request by ptersilie**
Runs test that randomly modify a given program and tests for exceptions.
If an exception occurs, or the modifications had unwanted side-effects, a
log is saved which can be used to create a stand-alone test.

### CODE REVIEW

File: lib/eco/fuzzytester.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+        self.treemanager.add_parser(self.parser, self.lexer, self.lang.name)
+
+    def setlang(self, lang):
+        self.lang = lang_dict[lang]
+        self.parser, self.lexer = lang_dict[lang].load()
+        self.treemanager = TreeManager()
+        self.treemanager.add_parser(self.parser, self.lexer, self.lang.name)
+
+    def move(self, direction, times):
+        for i in range(times): self.treemanager.cursor_movement(direction)
```
Comment (ltratt) [inline lib/eco/fuzzytester.py]: Split into two lines?

File: lib/eco/fuzzytester.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+        tree_compare, but less accurate."""
+        original = original.replace("\r", "").split("\n")
+        current = self.treemanager.export_as_text("/dev/null").replace("\r", "").split("\n")
+        for i in xrange(len(current)):
+            assert original[i] == current[i]
+
+    def next_node(self, node):
+        if node.children:
+            return node.children[0]
+        while(node.right_sibling() is None):
```
Comment (ltratt) [inline lib/eco/fuzzytester.py]: No brackets needed.

File: lib/eco/fuzzytester.py [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+                assert node1.next_term.symbol == node2.next_term.symbol
+            if isinstance(node1.symbol, MagicTerminal):
+                self.tree_compare(node1.symbol.ast, node2.symbol.ast)
+            if isinstance(node1, EOS) and isinstance(node2, EOS):
+                break
+            node1 = self.next_node(node1)
+            node2 = self.next_node(node2)
+
+    def get_random_key(self):
+        import random
```
Comment (ltratt) [inline lib/eco/fuzzytester.py]: I think this is already imported?

File: lib/eco/fuzzytester.py [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+
+    def random_insertdelete(self):
+        """Insert and delete random characters at random locations within a
+        program."""
+        print "Running random_insertdelete on {}".format(self.filename)
+        self.reset()
+
+        self.treemanager.import_file(self.program)
+        assert self.parser.last_status == True
+        #self.save()
```
Comment (ltratt) [inline lib/eco/fuzzytester.py]: Commented line.

**Comment by ptersilie** [comment]
Changes pushed!

**Comment by ltratt** [comment]
Please squash.

**Comment by ltratt** [comment]
Ping.

**Comment by ptersilie** [comment]
Squashed!

**Review by ltratt — APPROVED** [review]

**Pull request merged**